### DISCUSSION TITLE
Stage 2: User Interaction

### DISCUSSION THREAD

**Discussion by sanket1211996** [airavata-courses/SuperNova#22] [Show and tell]
I have created this discussion for first feature implementation prototype.

**Comment by sanket1211996** [comment]
Issue #16, I have implemented the social login using google from.

**Comment by sumedhsalvi** [comment]
I am working on the following prototype implementation: #20

**Comment by sanket1211996** [comment]
I am working on feature dashboard toolbar #23

**Reply by sanket1211996** [reply]
I have completed the dashboard toolbar, do review and merge the PR #24 @sumedhsalvi @richajha012

**Reply by sumedhsalvi** [reply]
I have reviewed and merged the PR #24

**Comment by richajha012** [comment]
I am currently working on user api get details endpoint. #26

**Reply by sanket1211996** [reply]
I have reviewed the PR #49  and merged. Go forward and close this issue #26.

**Comment by sumedhsalvi** [comment]
I am replacing the technology for gateway prototype implementation #20 from FastAPI to NodeJS because after trying out both approaches, nodeJS seems to be better as it provides more features for implementing gateway and is more flexible.
I will commit and raise PR as soon as I complete the initial stage.

**Reply by sanket1211996** [reply]
It's all good to try out different things. Let go with nodeJs one.

**Reply by sumedhsalvi** [reply]
I have implemented a NodeJS gateway prototype for the initial stage and raised PR #20 
Please review and merge.

**Comment by richajha012** [comment]
Issue #1 I have created user DB
Issue #34 I have created a user Db diagram as well

I have created PR for both of these issues

**Reply by sanket1211996** [reply]
Approved and merged PR #37 #38. Move the issue #1 #34 to done and close them.

**Comment by sumedhsalvi** [comment]
I have created PR for gateway API static route testing using fakeapi. #36 
Please review and merge. @sanket1211996 @richajha012

**Reply by sanket1211996** [reply]
Great work! I have merged the PR. Lets target for dynamic routes.

**Reply by sanket1211996** [reply]
You can close the issue and move it to done.

**Reply by sumedhsalvi** [reply]
Thanks! I am closing the issue.
Next, I am going to work on dynamics routing and registry part.

**Reply by sumedhsalvi** [reply]
I have created an issue for the dynamic routing and registry part. #45

**Comment by sanket1211996** [comment]
Please create PR and attach them to the restructuring issue #39  as discussed in the checkpoint meeting. I have created PR for the restructuring issue for angular project. #41 .

**Reply by sanket1211996** [reply]
Please do review my PR and merge. #41

**Reply by sumedhsalvi** [reply]
PR reviewed and merged. 
I have created PR for the Gateway restructuring. Please review and merge. #40

**Reply by sanket1211996** [reply]
#40 Reviewed and Merged.

**Reply by sanket1211996** [reply]
I have done the clean-up of ui files flown in gateway branch, created PR #55  for the same, issue #50  . Do review and merge.

**Comment by sanket1211996** [comment]
I have created an issue and branch #42 (Project1-Main Branch: Combine All Services View) .  In this branch, we can combine all dev-* branches on all individual services to have a combined view of all distributed services. This will only be used for visualization and will not be used for dev or deploy. Merging the dev branches into this for testing purpose.

**Comment by sumedhsalvi** [comment]
I have created PR for gateway API dynamic route testing and registry setup. #46 
Please review and merge. @sanket1211996 @richajha012

**Reply by richajha012** [reply]
I have reviewed merged .

**Reply by sumedhsalvi** [reply]
Thanks!
Now I am working on issue #47 - server registration endpoint to automatically register API with the gateway.

**Comment by sumedhsalvi** [comment]
I have created PR for the server registration endpoint. #48 
Please review and merge. @sanket1211996 @richajha012

**Reply by richajha012** [reply]
reviewed and merged.

**Reply by sumedhsalvi** [reply]
Thanks! closing the issue

**Comment by sanket1211996** [comment]
I have added User Interaction UI & Services #52 code. Please do review and merge. @sumedhsalvi @richajha012

**Reply by sumedhsalvi** [reply]
Reviewed and merged. You can close the issue.

**Comment by richajha012** [comment]
I have Created PR #68 for issue #58. Please review and merge.
